x/local-proxy/proxy/proxy.go

```diff
@@ -0,0 +1,13 @@
+package proxy
+
+type Proxy interface {
+	// GetAddr returns the address of the proxy server if it is configured to listen on a specific address.
+	// If address is not configured and the server is not started, it returns an empty string.
+	GetAddr() string
+	// StartServer starts the proxy server. If no address is specified, it will listen on a random localhost port
+	StartServer(addr string) error
```

Comment:
Let's make StartServer a standalone function that returns a Proxy object. (Similar how Listen returns a Listener). That way it will never be in an invalid state.

Reply:
Sounds nice. But I'm not sure about "never be in an invalid state" part. So, I can see `StartServer` (or better `StartProxy`) function returning a `Proxy` with interface like this:
```go
type Proxy interface {
	Addr() string // removing "Get" like in Listener
	Stop() error
	GetError() error // not sure if it should be "Error()", because it will implement an "error" interface
}
```
I can't come up with an example when things ca go wrong, but it seems for me that they can, and the proxy can broke for some reason. Will it be what you meant or is there something else I should do about never being in invalid state?

Reply:
Yes, something like that.